addons/binding/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVNotifyServlet.java

```diff
@@ -0,0 +1,193 @@
+            }
+            String path = request.getRequestURI();
+            logger.trace("Reqeust from {}:{}{} ({}, {})", ipAddress, request.getRemotePort(), path,
+                    request.getRemoteHost(), request.getProtocol());
+            if (!path.equalsIgnoreCase(PAIRING_NOTIFY_URI)) {
+                logger.error("Invalid request received - path = {}", path);
+                return;
+            }
+
+            if (data.contains(NOTIFY_PAIRING_CODE)) {
+                String deviceId = data.substring(data.indexOf("<uniqueDeviceID>") + "<uniqueDeviceID>".length(),
+                        data.indexOf("</uniqueDeviceID>"));
+                String pairingCode = data.substring(data.indexOf(NOTIFY_PAIRING_CODE) + NOTIFY_PAIRING_CODE.length(),
+                        data.indexOf("</messageBody>"));
+                logger.debug("Pairing code {} received for deviceID {}", pairingCode, deviceId);
+                if (!handlerFactory.notifyPairingResult(deviceId, ipAddress, pairingCode)) {
+                    logger.trace("Pairing data='{}'", data);
+                }
+            } else {
+                if (data.contains("STB_")) {
+                    data = data.replaceAll("&quot;", "\"");
+                    String stbMac = StringUtils.substringBetween(data, "<STB_Mac>", "</STB_Mac>");
+                    String stbEvent = "";
+                    if (data.contains("<STB_playContent>")) {
+                        stbEvent = StringUtils.substringBetween(data, "<STB_playContent>", "</STB_playContent>");
+                    } else if (data.contains("<STB_EitChanged>")) {
+                        stbEvent = StringUtils.substringBetween(data, "<STB_EitChanged>", "</STB_EitChanged>");
+                    } else {
+                        logger.debug("Unknown STB event: '{}'", data);
+                    }
+                    if (!stbEvent.isEmpty()) {
+                        if (!handlerFactory.notifyStbEvent(stbMac, stbEvent)) {
+                            logger.debug("Event not processed, data='{}'", data);
+                        }
+                    }
+                }
+            }
+
+        } catch (Exception e) {
+            if (data != null) {
+                logger.error("Exception: {} ({}), data='{}'", e.getMessage(), e.getClass(), data);
+            } else {
+                logger.error("Exception: {} ({})", e.getMessage(), e.getClass());
+            }
+        } finally {
+            setHeaders(resp);
+            resp.getWriter().write("");
+        }
+    }
+
+    @SuppressWarnings("resource")
+    private String inputStreamToString(HttpServletRequest request) throws IOException {
+        Scanner scanner = new Scanner(request.getInputStream()).useDelimiter("\\A");
+        return scanner.hasNext() ? scanner.next() : "";
+    }
+
+    private void setHeaders(HttpServletResponse response) {
+        response.setCharacterEncoding(CHARSET_UTF8);
+        // response.setContentType(CONTENT_TYPE_XML);
```

Comment:
Please remove if not required or add a comment why it is here

Reply:
done